dkc/core/rest/folder.py

```diff
@@ -43,14 +42,31 @@ class Meta:
             'public',
             'access',
         ]
+        # ModelSerializer cannot auto-generate validators for model-level constraints
+        validators = [
+            serializers.UniqueTogetherValidator(
+                queryset=Folder.objects.all(),
+                fields=['parent', 'name'],
+                message='Folder names must be unique among siblings.',
+            ),
+            # This could also be implemented as a UniqueValidator on 'name',
+            # but its easier to not explicitly redefine the whole serializer field
+            serializers.UniqueTogetherValidator(
+                queryset=Folder.objects.filter(parent=None),
+                fields=['name'],
+                message='Root folders must have a unique name.',
+            ),
+            # folder_max_depth and unique_root_folder_per_tree are internal sanity constraints,
+            # and do not need to be enforced as validators
+        ]
```

Comment:
It looks like this doesn't yet capture the logic that's in `Folder.clean`: checking that there's also no `File` with the same name.

Reply:
You're right, I missed that. I'll try to fix it and add a test.